src/test/java/com/commercetools/sync/sdk2/customobjects/CustomObjectSyncTest.java

```diff
@@ -0,0 +1,582 @@
+
+    final CustomObject existingCustomObject = mock(CustomObject.class);
+    when(existingCustomObject.getContainer()).thenReturn("someContainer");
+    when(existingCustomObject.getKey()).thenReturn("otherKey");
+    when(existingCustomObject.getValue()).thenReturn(JsonNodeFactory.instance.numberNode(2020));
+
+    final CustomObject updatedCustomObject = mock(CustomObject.class);
+    when(updatedCustomObject.getContainer()).thenReturn("someContainer");
+    when(updatedCustomObject.getKey()).thenReturn("someKey");
+    when(updatedCustomObject.getValue()).thenReturn(2020);
+
+    final Set<CustomObject> existingCustomObjectSet = new HashSet<CustomObject>();
+    existingCustomObjectSet.add(existingCustomObject);
+
+    final CustomObjectService customObjectService = mock(CustomObjectService.class);
+    when(customObjectService.fetchMatchingCustomObjects(anySet()))
+        .thenReturn(completedFuture(existingCustomObjectSet));
+    when(customObjectService.upsertCustomObject(any()))
+        .thenReturn(completedFuture(Optional.of(updatedCustomObject)));
+
+    final CustomObjectDraft newCustomObjectDraft =
+        CustomObjectDraftBuilder.of().container("someContainer").key("someKey").value(2020).build();
+
+    // test
+    CustomObjectSyncStatistics syncStatistics =
+        new CustomObjectSync(spyCustomObjectSyncOptions, customObjectService)
+            .sync(singletonList(newCustomObjectDraft))
+            .toCompletableFuture()
+            .join();
+
+    // assertion
+    assertThat(exceptions).hasSize(0);
+    assertThat(errorMessages).hasSize(0);
+    assertAll(
+        () -> assertThat(syncStatistics.getProcessed().get()).isEqualTo(1),
+        () -> assertThat(syncStatistics.getCreated().get()).isEqualTo(1),
+        () -> assertThat(syncStatistics.getUpdated().get()).isEqualTo(0),
+        () -> assertThat(syncStatistics.getFailed().get()).isEqualTo(0));
+    verify(spyCustomObjectSyncOptions).applyBeforeCreateCallback(newCustomObjectDraft);
+  }
+
+  @Test
+  void sync_WitEmptyValidDrafts_ShouldFailed() {
+    final List<String> errorMessages = new ArrayList<>();
+    final List<Throwable> exceptions = new ArrayList<>();
+
+    final CustomObjectSyncOptions spyCustomObjectSyncOptions =
+        initCustomObjectSyncOptions(errorMessages, exceptions);
+
+    final CustomObjectService customObjectService = mock(CustomObjectService.class);
+    when(customObjectService.fetchMatchingCustomObjects(anySet()))
+        .thenReturn(completedFuture(emptySet()));
+    when(customObjectService.upsertCustomObject(any()))
+        .thenReturn(completedFuture(Optional.empty()));
+
+    // test
+    CustomObjectSyncStatistics syncStatistics =
+        new CustomObjectSync(spyCustomObjectSyncOptions, customObjectService)
+            .sync(singletonList(null))
+            .toCompletableFuture()
+            .join();
+
+    // assertion
+    assertThat(exceptions).hasSize(1);
```

Comment:
The exceptions indeed has 1 element, but this element is null. Please verify.

Reply:
[8d23794](https://github.com/commercetools/commercetools-sync-java/pull/997/commits/8d23794300064b1bd7ef3af85823c10dca4561d7)

Reply:
So null exception is expected? It's better to not pass any exceptions at all.

Reply:
Ah, now i got the point.... Checkin...

Reply:
[855d2dc](https://github.com/commercetools/commercetools-sync-java/pull/997/commits/855d2dcee69fa05066d328d155138cccbd87b7da)